File: public/javascript/views/cards.js
var Cards = Backbone.View.extend({
  initialize: function() {
    this.model = new Deck();
    this.listenTo(this.model, 'sync', this.render);
    this.model.fetch();
    eventNeo.on('showCard', this.showCard);
  },
  events: {
    'click #cardCreateButton': 'createCard'
  },
  render: function() {
    $(this.el).html(cardsTemplate(this.processDeck()));
    this.addEvents();
    this.$el = $(this.el);
    return this;
  },
  addEvents: function() {
    var data = this.model.models;
    var that = this;
    for(var i = 0; i < data.length; i++) {
      (function() {
        var card = data[i].attributes;
        $('#card-' + card.id).click(function() {eventNeo.trigger('showCard', card, that)} );
      }());
    }
  },
  showCard: function(card, that) {
    cardView = new Card(card);
    cardView.render();
    that.$el.html(cardView.$el.html());
  },
  processDeck: function() {
    var deck = {
      ready: [],
      progress: [],
      done: []
    };
    var data = this.model.models;
    for(var i = 0; i < data.length; i++) {
      var card = data[i].attributes;
      card.id = i;
      if(card.stack == 'ready') {
        deck.ready.push(card);
      } else if (card.stack == 'progress') {
        deck.progress.push(card);
      } else {
        deck.done.push(card);
      }
    }
    return deck;
  },
  createCard: function(e) {
    e.preventDefault();
    this.model.create({
      username: 'x',
      stack: 'ready',
      title: $('#cardTitle').val(),
      body: $('#cardBody').val(),
      acceptance: $('#cardAcceptance').val()
    });
  }
})

File: lib/behaviors/cards.rb
class Cards
  def Cards.getDeck(params, session)
    Card.where(username: session[:username])
  end

  def Cards.createCard(body_as_hash, session)
    validateCreateCard(body_as_hash, session)
    card = Card.new
    card.stack = 'ready'
    card.username = session[:username]
    card.title = body_as_hash['title'].to_s
    card.body = body_as_hash['body'].to_s
    card.acceptance = body_as_hash['acceptance'].to_s
    card.save!
    return card.attributes
  end

  def Cards.promoteCard(params, session)

  end

  def Cards.validateCreateCard(body_as_hash, session)
    if session[:username].to_s == ''
      raise StandardError
    end
    if body_as_hash['title'].to_s == ''
      raise StandardError
    end
    if body_as_hash['body'].to_s == ''
      raise StandardError
    end
    if body_as_hash['acceptance'].to_s == ''
      raise StandardError
    end
  end

  def validatePromoteCard()

  end
end
